### PR TITLE
Validate the salesforce host

### PR DESCRIPTION
**Pull request by jsteel**

### CODE REVIEW

File: lib/services/salesforce.rb [hunk 1]
```diff
@@ -11,20 +11,28 @@ class AhaServices::Salesforce < AhaService
...
-    client.post('/services/apexrest/ahaapp/aha_rest_api/settings', 
-      idea_portal_url: data.idea_portal_url, 
-      jwt_secret_key: data.jwt_secret_key)
-  rescue Exception => e
-    logger.debug("Salesforce authentication problem #{e.class}: #{e.message} #{e.backtrace.join("\n")}")
-    if e.message.include? 'The REST API is not enabled for this Organization.'
-      raise ConfigurationError, "The REST API is not enabled for this Organization."
-    else
-      raise ConfigurationError, "Authentication failed. Please use the 'Authenticate' button to connect to Salesforce."
+    validate_host or raise ConfigurationError, "Please ensure the custom host is a host ending in salesforce.com."
```
Comment (schneidmaster) [inline lib/services/salesforce.rb]: I think we should make this error message more clear, a user who entered `https://myurl.salesforce.com` probably does not understand from this message that a "host" doesn't include the `https://` part. I see we're putting it in the error logs but it seems to me that it should show in the error msg as well. I think it'd be fine if we returned the error strings being logged in `validate_host` and then made the error message `"Invalid host: #{error_msg}"` (or we could just raise the `ConfigurationError` directly within `validate_host`)

Comment (jsteel) [inline lib/services/salesforce.rb]: Agreed. I didn't really like that error message. How about this one? Also, the logger.errors are more visible than the actual raised error I think.
screen shot 2017-08-22 at 4 31 37 pm

Comment (jsteel) [inline lib/services/salesforce.rb]: The error is different than shown in the image.

File: lib/services/salesforce.rb [hunk 2]
```diff
@@ -11,20 +11,30 @@ class AhaServices::Salesforce < AhaService
...
-    client.post('/services/apexrest/ahaapp/aha_rest_api/settings', 
-      idea_portal_url: data.idea_portal_url, 
-      jwt_secret_key: data.jwt_secret_key)
-  rescue Exception => e
-    logger.debug("Salesforce authentication problem #{e.class}: #{e.message} #{e.backtrace.join("\n")}")
-    if e.message.include? 'The REST API is not enabled for this Organization.'
-      raise ConfigurationError, "The REST API is not enabled for this Organization."
-    else
-      raise ConfigurationError, "Authentication failed. Please use the 'Authenticate' button to connect to Salesforce."
+    validate_host or raise ConfigurationError, "Please ensure the custom host ends in salesforce.com and do not include http:// or https://."
```
Comment (schneidmaster) [inline lib/services/salesforce.rb]: Tiny complaint, can we make this `does not include` instead of `do not include`? (Flows better grammatically)

Comment (jsteel) [inline lib/services/salesforce.rb]: Sold.

**Pull request merged**